pkg/imagefilter/formatter.go

```diff
@@ -27,20 +29,28 @@ type ResultsFormatter interface {
 	Output(io.Writer, []Result) error
 }
 
-// NewResultFormatter will create a formatter based on the given format.
+var supportedFormatterMap = map[string]ResultsFormatter{
+	string(OutputFormatDefault):   &textResultsFormatter{},
+	string(OutputFormatText):      &textResultsFormatter{},
+	string(OutputFormatJSON):      &jsonResultsFormatter{},
+	string(OutputFormatTextShell): &shellResultsFormatter{},
+	string(OutputFormatTextShort): &textShortResultsFormatter{},
+}
+
+// SupportedOutputFormats returns a list of supported output formats
+func SupportedOutputFormats() []string {
+	keys := maps.Keys(supportedFormatterMap)
+	sort.Strings(keys)
```

Comment:
Just a question; but `maps.Keys` returns an iterator. Which part of this function turns that into a slice, is it the `return` or the `sort.Strings`?

Reply:
This is a bit of a funny one, the new "maps" package in 1.23 does return an iterator but the compat package golang.org/x/exp/maps (that we need to use as we are not on 1.23 yet) just returns a []string slice. We can keep the compat maps until it gets removed and then we either port to iterators or just implement our own .Keys() (its trivial enough).